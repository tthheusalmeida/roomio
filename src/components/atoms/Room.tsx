import { mergeClassNames } from "@/utils/classNames";

interface CountdownTimerProps {
  label: string;
  className?: string;
}

export default function Room({ label, className }: CountdownTimerProps) {
  return (
    <h4
      className={mergeClassNames(
        "w-full h-11 flex items-center justify-center font-medium text-base bg-violet-900 text-violet-100 rounded-lg",
        className
      )}
    >
      {label}
    </h4>
  );
}
