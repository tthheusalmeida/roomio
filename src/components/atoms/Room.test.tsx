import { render, screen } from "@testing-library/react";
import Room from "./Room";

describe("Room component", () => {
  it("renders the label text correctly", () => {
    const { asFragment } = render(<Room label="10:00" />);

    const heading = screen.getByText("10:00");
    expect(heading).toBeInTheDocument();
    expect(heading.tagName).toBe("H4");

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies extra classes passed via className", () => {
    const { asFragment } = render(
      <Room label="10:00" className="extra-class" />
    );

    const heading = screen.getByText("10:00");
    expect(heading).toHaveClass("extra-class");

    expect(asFragment()).toMatchSnapshot();
  });
});
