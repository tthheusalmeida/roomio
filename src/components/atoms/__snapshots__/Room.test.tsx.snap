// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Room component applies extra classes passed via className 1`] = `
<DocumentFragment>
  <h4
    class="w-full h-11 flex items-center justify-center font-medium text-base bg-violet-900 text-violet-100 rounded-lg extra-class"
  >
    10:00
  </h4>
</DocumentFragment>
`;

exports[`Room component renders the label text correctly 1`] = `
<DocumentFragment>
  <h4
    class="w-full h-11 flex items-center justify-center font-medium text-base bg-violet-900 text-violet-100 rounded-lg"
  >
    10:00
  </h4>
</DocumentFragment>
`;
